httplog/httplog.go

```diff
@@ -37,6 +39,66 @@ type Logger struct {
 	mode Mode
 }
 
+type Config struct {
+	ModeOptions []map[string]Mode
+	defaults    map[string]Mode
+}
+
+func ParseMode(s string) (map[string]Mode, error) {
+	var modeStr string
+	var mode Mode
+	module := ""
+
+	if strings.Contains(s, ":") {
+		m := strings.SplitN(s, ":", 2)
+		module = m[0]
+		modeStr = m[1]
+	} else {
+		modeStr = s
+	}
+
+	switch modeStr {
+	case "none":
+		mode = None
+	case "short-url":
+		mode = ShortURL
+	case "url":
+		mode = URL
+	case "headers":
+		mode = Headers
+	case "body":
+		mode = Body
+	case "errors":
+		mode = Errors
+	default:
+		return map[string]Mode{}, fmt.Errorf("unknown HTTP log mode: %s", s)
+	}
+
+	return map[string]Mode{module: mode}, nil
```

Comment:
Returning map containing single element makes little sense.
Parse should return `ModuleMode` as @Choraden suggested.